crates/compute/src/alloc.rs

```diff
@@ -17,6 +17,12 @@ pub trait ComputeAllocator<F, Mem: ComputeMemory<F>> {
 	/// - `n` must be a multiple of `Mem::MIN_SLICE_LEN`
 	fn alloc(&self, n: usize) -> Result<Mem::FSliceMut<'_>, Error>;
 
+	/// Allocates the remaining elements in the slice
+	///
+	/// This allows another allocator to have unique mutable access to the rest of the elements in
+	/// this allocator until it gets dropped, at which point this allocator can be used again
+	fn remaining(&mut self) -> &mut Mem::FSliceMut<'a>;
```

Comment:
This should not return an `'a` lifetime, it needs to return `'_`. That is a fancy way of doing:

```rust
fn remaining<'a>(&'a mut self) -> &mut Mem::FSliceMut<'a>;
```